lib/data/attachments.js

```diff
@@ -33,14 +34,24 @@ const streamAttachments = (inStream, decryptor) => {
     write(x, _, done) {
       const att = x.row;
 
-      // this sanitization means that two filenames could end up identical.
-      // luckily, this is not actually illegal in the zip spec; two files can live at precisely
-      // the same location, and the conflict is dealt with interactively by the unzipping client.
-      const content = (att.localKey == null)
-        ? att.content
-        : decryptor(att.content, att.keyId, att.localKey, att.instanceId, att.index);
+      const appendContent = (rawContent) => {
+        // this sanitization means that two filenames could end up identical.
+        // luckily, this is not actually illegal in the zip spec; two files can live at precisely
+        // the same location, and the conflict is dealt with interactively by the unzipping client.
+        const content = (att.localKey == null)
+          ? rawContent
+          : decryptor(rawContent, att.keyId, att.localKey, att.instanceId, att.index);
 
-      archive.append(content, { name: join('media', processName(att.name)) }, done);
+        archive.append(content, { name: join('media', processName(att.name)) }, done);
+      };
+
+      if (att.s3_status === 'uploaded') {
+        s3.getContentFor(att)
+          .then(appendContent)
+          .catch(err => { this.destroy(err); });
```

Comment:
I see what you mean about this query: 
> would it be preferable to make blob.content async everywhere, and moving the s3 check/fetch inside the getter?
> * pro: neater code, less likely to have bugs where blob.content used when data is not available
> * con: seems intrusive of frames

(Just highlighting this one because it seemed like the smallest example.)

Maybe it's okay to leave it like this since there are only about 4 places blobs are used (unless you have an idea for some clever alternative):
1. exporting attachments (here)
2. exporting encrypting submissions that were stored as blobs
3. processing client audit attachments
4. (different mechanism, `urlForBlob` instead of `getContentFor` line the ones above) the http blob response 

As long as each of these 4 paths is tested, since each might have a different behavior if there is an error fetching from S3.

Some errors i'm thinking of:
- auth problem
- data doesn't exist where it should
- response from s3 is taking too long
- data is broken in some other way?